test/provision/manifest/dns_deployment.yaml

```diff
@@ -58,78 +43,58 @@ metadata:
-    addonmanager.kubernetes.io/mode: Reconcile
+  name: kube-dns
+  namespace: kube-system
 spec:
-  # replicas: not specified here:
-  # 1. In order to make Addon Manager do not reconcile this replicas parameter.
-  # 2. Default is 1.
-  # 3. Will be tuned in real time if DNS horizontal auto-scaling is turned on.
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: kube-dns
   strategy:
     rollingUpdate:
       maxSurge: 10%
       maxUnavailable: 0
-  selector:
-    matchLabels:
-      k8s-app: kube-dns
+    type: RollingUpdate
   template:
     metadata:
+      creationTimestamp: null
       labels:
         k8s-app: kube-dns
-      annotations:
-        scheduler.alpha.kubernetes.io/critical-pod: ''
     spec:
-      tolerations:
-      - key: "CriticalAddonsOnly"
-        operator: "Exists"
-      volumes:
-      - name: kube-dns-config
-        configMap:
-          name: kube-dns
-          optional: true
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - amd64
       containers:
-      - name: kubedns
-        image: k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.10
-        resources:
-          # TODO: Set memory limits when we've profiled the container for large
-          # clusters, then set request = limit to keep this container in
-          # guaranteed class. Currently, this container falls into the
-          # "burstable" category so the kubelet doesn't backoff from restarting it.
-          limits:
-            memory: 170Mi
-          requests:
-            cpu: 100m
-            memory: 70Mi
+      - args:
+        - --domain=cluster.local.
+        - --dns-port=10053
+        - --config-dir=/kube-dns-config
+        - --v=5
+        env:
+        - name: PROMETHEUS_PORT
+          value: "10055"
+        image: gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.10
+        imagePullPolicy: IfNotPresent
         livenessProbe:
+          failureThreshold: 5
           httpGet:
             path: /healthcheck/kubedns
             port: 10054
             scheme: HTTP
           initialDelaySeconds: 60
-          timeoutSeconds: 5
+          periodSeconds: 10
           successThreshold: 1
-          failureThreshold: 5
-        readinessProbe:
-          httpGet:
-            path: /readiness
-            port: 8081
-            scheme: HTTP
-          # we poll on pod startup for the Kubernetes master service and
-          # only setup the /readiness HTTP server once that's available.
-          initialDelaySeconds: 3
           timeoutSeconds: 5
-        args:
-        - --domain=cluster.local.
-        - --dns-port=10053
-        - --config-dir=/kube-dns-config
-        - --v=5
-        env:
-        - name: PROMETHEUS_PORT
-          value: "10055"
+        name: kubedns
```

Comment:
One thing that would make this easier to review is maintain the ordering of the fields in the YAML. For instance, the ordering of `-args` and `name` of the container is different, etc., and makes the diff larger than it has to be.

Reply:
I get the changes from the default kubeadm and update the version, so this is why the order change.